kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericMatrix.js

```diff
@@ -0,0 +1,308 @@
+                });
+        },
+
+        buildColumnTableData: function(){
+            var matrixStat = this.matrixStat;
+            var tableData = [];
+            for(var i = 0; i < matrixStat.column_descriptors.length; i++){
+                var desc = matrixStat.column_descriptors[i];
+                var stat = matrixStat.column_stats[i];
+                tableData.push({
+                    index: desc.index,
+                    id: desc.id,
+                    name: desc.name,
+                    min: stat.min ? stat.min.toFixed(2) : null,
+                    max: stat.max ? stat.max.toFixed(2) : null,
+                    avg: stat.avg ? stat.avg.toFixed(2) : null,
+                    std: stat.std ? stat.std.toFixed(2) : null,
+                    missing_values: stat.missing_values ? 'Yes' : 'No'
+                });
+            }
+            return tableData;
+        },
+
+        buildRowTableData: function(){
+            var matrixStat = this.matrixStat;
+            var tableData = [];
+            for(var i = 0; i < matrixStat.row_descriptors.length; i++){
+                var desc = matrixStat.row_descriptors[i];
+                var stat = matrixStat.row_stats[i];
+                tableData.push(
+                    {
+                        index: desc.index,
+                        id: desc.id,
+                        name: desc.name,
+                        min: stat.min ? stat.min.toFixed(2) : null,
+                        max: stat.max ? stat.max.toFixed(2) : null,
+                        avg: stat.avg ? stat.avg.toFixed(2) : null,
+                        std: stat.std ? stat.std.toFixed(2) : null,
+                        missing_values: stat.missing_values ? 'Yes' : 'No'
+                    }
+                );
+            }
+            return tableData;
+        },
+
+        makeRow: function(name, value) {
+            var $row = $('<tr/>')
+                .append($('<th />').css('width','20%').append(name))
+                .append($('<td />').append(value));
+            return $row;
+        },
+
+        getData: function() {
+            return {
+                type: 'ExpressionMatrix',
+                id: this.options.matrixID,
+                workspace: this.options.workspaceID,
+                title: 'Expression Matrix'
+            };
+        },
+
+        loading: function(isLoading) {
+            if (isLoading) {
+                this.showMessage('<img src=\'' + this.options.loadingImage + '\'/>');
+            } else {
+                this.hideMessage();
+            }
+        },
+
+        showMessage: function(message) {
+            var span = $('<span/>').append(message);
+
+            this.$messagePane.append(span);
+            this.$messagePane.show();
+        },
+
+        hideMessage: function() {
+            this.$messagePane.hide();
+            this.$messagePane.empty();
+        },
+
+        uuid: function() {
```

Comment:
I'd argue that this is just wrapping a single line call, and it's only used once, so it could probably be removed.